### PR TITLE
[glslang] Update to version 8.13.3559

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
- Use upstream target install support.
- Add a package feature for spirv-tools-opt integration.

I'm currently preparing a cmake integrated bgfx port (requested here #1303) which requires a newer `glslang` version and `spirv-tools-opt` integration.

Side note: The upstream install targets are not namespaced as opposed to those previously patched in.

**These changes depend on** #10441

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@BurningEnlightenment, thanks for your contribution!

Let's waiting for PR https://github.com/microsoft/vcpkg/pull/10441 merge first.

**Comment by dan-shaw** [comment]
Just merged #10441

**Comment by BurningEnlightenment** [comment]
I'm absolutely clueless w.r.t. the OSX build failure. From the logs:
```
[22/32] : && /Library/Developer/CommandLineTools/usr/bin/c++ -fPIC -Wimplicit-fallthrough -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 1.0.0 -o libshaderc/libshaderc_shared.1.dylib -install_name @rpath/libshaderc_shared.1.dylib libshaderc/CMakeFiles/shaderc_shared.dir/src/shaderc.cc.o  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libglslangd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOSDependentd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOGLCompilerd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libglslangd.a libshaderc_util/libshaderc_util.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRVd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRV-Tools.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libglslangd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOSDependentd.a -lThreads::Threads /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOGLCompilerd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libHLSLd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRV-Tools-opt.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRV-Tools.a && :
FAILED: libshaderc/libshaderc_shared.1.dylib 
: && /Library/Developer/CommandLineTools/usr/bin/c++ -fPIC -Wimplicit-fallthrough -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 1.0.0 -o libshaderc/libshaderc_shared.1.dylib -install_name @rpath/libshaderc_shared.1.dylib libshaderc/CMakeFiles/shaderc_shared.dir/src/shaderc.cc.o  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libglslangd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOSDependentd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOGLCompilerd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libglslangd.a libshaderc_util/libshaderc_util.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRVd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRV-Tools.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libglslangd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOSDependentd.a -lThreads::Threads /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libOGLCompilerd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libHLSLd.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRV-Tools-opt.a /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libSPIRV-Tools.a && :
ld: library not found for -lThreads::Threads
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
`-lThreads::Threads` indicates that the `FindThreads` cmake module which is used by the upstream project [here](https://github.com/google/shaderc/blob/eb7bd643ef43ffb9ba091e39a389d0a2a923b1aa/libshaderc/CMakeLists.txt#L49-L55) doesn't work/isn't used correctly. However, I can't tell what the underlying issue is and whether upstream did something wrong.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@BurningEnlightenment, thanks for the PR! 

I noticed that glslang depends on spirv-tools, it seems that spirv-tools failed with arm-uwp and x64-uwp, could you add the 'Supports' and vcpkg_fail_port_install to spirv-tools port as well?


https://github.com/microsoft/vcpkg/blob/b3bdac11f0b25544fbedc7c6913f0f1d16404aa2/scripts/ci.baseline.txt#L1680

File: ports/shaderc/portfile.cmake [hunk 1]
```diff
@@ -31,23 +27,23 @@ get_filename_component(PYTHON3_EXE_PATH ${PYTHON3} DIRECTORY)
...
+    SOURCE_PATH "${SOURCE_PATH}"
     PREFER_NINJA
-    OPTIONS -DSHADERC_SKIP_TESTS=true ${OPTIONS} -Dglslang_SOURCE_DIR=${CURRENT_INSTALLED_DIR}/include -Dspirv-tools_SOURCE_DIR=${CURRENT_INSTALLED_DIR}/include 
-    OPTIONS_DEBUG -DSUFFIX_D=true
-    OPTIONS_RELEASE -DSUFFIX_D=false
+    OPTIONS
+        -DSHADERC_SKIP_TESTS=true
+        ${SHADERC_CRT_OPTIONS}
+        -Dglslang_SOURCE_DIR=""
+        -Dspirv-tools_SOURCE_DIR=""
```
[the remark this answers is not in the document]
Comment (BurningEnlightenment) [reply ports/shaderc/portfile.cmake]: Yes, those variables specify the path to the normally vendored dependencies. By default they point to e.g. `${SOURCE_PATH}/third_party/glslang`. The include path for those packages is computed relative to these. Therefore setting these to empty adds the correct include path.

File: ports/shaderc/install-targets.patch [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ 
+@@ -40,10 +41,7 @@ if(SHADERC_ENABLE_INSTALL)
+     DESTINATION
+       ${CMAKE_INSTALL_INCLUDEDIR}/shaderc)
+ 
+-  install(TARGETS shaderc shaderc_shared
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
++  install(TARGETS shaderc EXPORT shaderc-targets)
```
Comment (JackBoosY) [inline ports/shaderc/install-targets.patch]: This modification will cause the libraries and binaries generated by shader to not be installed. Is it reasonable?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by PhoebeHui** [comment]
@BurningEnlightenment, could you address the review request from Jack, thanks!

**Comment by LilyWangL** [comment]
Pinging @BurningEnlightenment for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**